flight/PiOS/Common/pios_bmp280.c

```diff
@@ -540,37 +526,38 @@ int32_t PIOS_BMP280_Test()
 	return 0;
 }
 
-static void PIOS_BMP280_Task(void *parameters)
+static bool PIOS_BMP280_callback(void *ctx, void *output,
+		int ms_to_wait, int *next_call)
 {
-	int32_t read_adc_result = 0;
-
-	PIOS_BMP280_StartADC();
+	struct bmp280_dev *dev = (struct bmp280_dev *)ctx;
 
-	struct pios_sensor_baro_data data = { };
+	PIOS_Assert(dev);
+	PIOS_Assert(output);
 
-	while (1) {
-		// Poll a bit faster than sampling rate
-		PIOS_Thread_Sleep(PIOS_BMP280_GetDelay() * 3 / 5);
-		PIOS_BMP280_ClaimDevice();
-		read_adc_result = PIOS_BMP280_ReadADC();
-		PIOS_BMP280_ReleaseDevice();
+	// Poll a bit faster than sampling rate
+	*next_call = PIOS_BMP280_GetDelay() * 3 / 5;
 
-		if (read_adc_result) {
-			continue;
-		}
+	int32_t read_adc_result = 0;
 
-		// Compute the altitude from the pressure and temperature and send it out
+	PIOS_BMP280_StartADC();
 
-		data.pressure = ((float) dev->compensatedPressure) / 256.0f / 1000.0f;
+	PIOS_BMP280_ClaimDevice();
+	read_adc_result = PIOS_BMP280_ReadADC();
+	PIOS_BMP280_ReleaseDevice();
 
-		float calc_alt = 44330.0f * (1.0f - powf(data.pressure / BMP280_P0, (1.0f / 5.255f)));
+	if (read_adc_result) {
+		return false;
+	}
 
-		data.temperature = ((float) dev->compensatedTemperature) / 256.0f / 100.0f;
+	struct pios_sensor_baro_data *data = (struct pios_sensor_baro_data *)output;
```

Comment:
What's the purpose of this var? Doesn't seem to add anything.

Reply:
Right, should probably put it directly in the function header.

Reply:
Ahhh, it's for the cast. No, it's fine as is. 👍